pkg/loader/model.go

```diff
@@ -167,94 +167,75 @@ func (dml *DML) oldPrimaryKeyValues() []interface{} {
 
 // TableName returns the fully qualified name of the DML's table
 func (dml *DML) TableName() string {
+	if dml.DestDBType == "oracle" {
+		return fmt.Sprintf("%s.%s", dml.Database, dml.Table)
+	}
 	return quoteSchema(dml.Database, dml.Table)
 }
 
-// OracleTableName returns the fully qualified name of the DML's table in oracle db
-func (dml *DML) OracleTableName() string {
-	return fmt.Sprintf("%s.%s", dml.Database, dml.Table)
-}
-
 func (dml *DML) updateSQL() (sql string, args []interface{}) {
 	builder := new(strings.Builder)
 
 	fmt.Fprintf(builder, "UPDATE %s SET ", dml.TableName())
-
+	colName := ""
+	oracleHolderPos := 1
 	for _, name := range dml.columnNames() {
 		if len(args) > 0 {
 			builder.WriteByte(',')
 		}
 		arg := dml.Values[name]
-		fmt.Fprintf(builder, "%s = ?", quoteName(name))
-		args = append(args, arg)
-	}
-
-	builder.WriteString(" WHERE ")
-
-	whereArgs := dml.buildWhere(builder)
-	args = append(args, whereArgs...)
-
-	builder.WriteString(" LIMIT 1")
-	sql = builder.String()
-	return
-}
-
-func (dml *DML) oracleUpdateSQL() (sql string) {
-	builder := new(strings.Builder)
-
-	fmt.Fprintf(builder, "UPDATE %s SET ", dml.OracleTableName())
-
-	for i, name := range dml.columnNames() {
-		if i > 0 {
-			builder.WriteByte(',')
+		if dml.DestDBType == "oracle" {
+			colName = escapeName(name)
+		} else {
+			colName = quoteName(name)
 		}
-		value := dml.Values[name]
-		if value == nil {
-			fmt.Fprintf(builder, "%s = NULL", escapeName(name))
+		if dml.DestDBType == "oracle" {
+			fmt.Fprintf(builder, "%s = :%d", colName, oracleHolderPos)
+			oracleHolderPos++
 		} else {
-			fmt.Fprintf(builder, "%s = %s", escapeName(name), genOracleValue(dml.UpColumnsInfoMap[name], value))
+			fmt.Fprintf(builder, "%s = ?", colName)
 		}
```

Comment:
Can we directly write here? `colName` seems useless here but will take another copy time.

Reply:
Yes